core/src/main/java/tc/oc/pgm/scoreboard/SidebarMatchModule.java

```diff
@@ -221,10 +225,39 @@ private boolean isBlitz() {
     return match.getModule(BlitzMatchModule.class) != null;
   }
 
+  // Determines if wool objectives should be given their own rows, or all shown on 1 row.
   private boolean isCompactWool() {
     WoolMatchModule wmm = match.getModule(WoolMatchModule.class);
     return wmm != null
-        && MAX_ROWS < wmm.getWools().keySet().size() * 2 - 1 + wmm.getWools().values().size();
+        && !(wmm.getWools().keySet().size() * 2 - 1 + wmm.getWools().values().size() < MAX_ROWS);
+  }
+
+  // Determines if all the map objectives can fit onto the scoreboard with empty rows in between.
+  private boolean isSuperCompact(Set<Competitor> competitorsWithGoals) {
+    int rowsUsed = competitorsWithGoals.size() * 2 - 1;
+
+    WoolMatchModule wmm = match.getModule(WoolMatchModule.class);
+    if (wmm != null) {
+      if (isCompactWool()) {
+        rowsUsed += wmm.getWools().keySet().size() * 2 - 1;
+      } else {
+        rowsUsed += wmm.getWools().values().size();
+      }
+    }
+
+    ControlPointMatchModule cpmm = match.getModule(ControlPointMatchModule.class);
+    if (cpmm != null) rowsUsed += cpmm.getControlPoints().size();
+
+    FlagMatchModule fmm = match.getModule(FlagMatchModule.class);
+    if (fmm != null) rowsUsed += fmm.getFlags().size();
+
+    CoreMatchModule cmm = match.getModule(CoreMatchModule.class);
+    if (cmm != null) rowsUsed += cmm.getCores().size();
+
+    DestroyableMatchModule dmm = match.getModule(DestroyableMatchModule.class);
+    if (dmm != null) rowsUsed += dmm.getDestroyables().size();
```

Comment:
I think you could use `GoalMatchModule` to get all the goals without checking each module.

Reply:
Yeah, I see that would be the easy way. Sorry, I misunderstood how goals work.